src/web/src/components/SearchResults.tsx

```diff
@@ -55,14 +55,16 @@ const useStyles = makeStyles((theme: Theme) =>
 
 const SearchResults = () => {
   const classes = useStyles();
-  const { textParam, filter, toggleHelp } = useSearchValue();
+  const { author, post, title, from, to, toggleHelp } = useSearchValue();
   const [totalPosts, setTotalPosts] = useState(0);
 
   const prepareUrl = (index: number) =>
-    `${searchServiceUrl}?text=${encodeURIComponent(textParam)}&filter=${filter}&page=${index}`;
+    `${searchServiceUrl}/?author=${encodeURIComponent(author)}&post=${encodeURIComponent(
+      post
+    )}&title=${encodeURIComponent(title)}${from ? `&from=${from}` : ''}${to ? `&to=${to}` : ''}`;
 
   // We only bother doing the request if we have something to search for.
-  const shouldFetch = () => textParam.length > 0;
+  const shouldFetch = () => post.length > 0 || author.length > 1;
```

Comment:
This is causing the issue I mentioned btw, need to modify this so that the actual object is not being updated but the param itself.